### PR TITLE
chore(gh): use golang-ci lint directly in sanity workflow

### PR DESCRIPTION
**Pull request by njhale**
Use golang-ci lint directly in the "sanity" GitHub Workflow. This
avoids the use of a third-party golangci-lint action which can
sporadically timeout, causing the workflow to fail.

### CODE REVIEW

File: pkg/controller/registry/grpc/source.go [hunk 1]
```diff
@@ -147,7 +147,8 @@ func getEnvAny(names ...string) string {
 }
 
 func grpcConnection(address string) (*grpc.ClientConn, error) {
-	dialOptions := []grpc.DialOption{grpc.WithInsecure()}
+	creds := insecure.NewCredentials()
```
Comment (dinhxuanvu) [inline pkg/controller/registry/grpc/source.go:150]: Are these changes required because of what the lint took found?

**Comment by tylerslaton** [comment]
Looks like it needs a rebase and then I'll go ahead and LGTM it, we had to do something similar for Combo.

**Comment by dinhxuanvu** [comment]
The sanity test still fails due to timeout.

**Comment by exdx** [comment]
Surprising that this change causes such a large diff in our dependencies, but I guess it's fine 🤷

File: pkg/controller/registry/grpc/source.go (hunk 1 above)
Comment (njhale) [reply pkg/controller/registry/grpc/source.go:150]: yup

**Comment by njhale** [comment]
/hold

This is still timing out. Going to do some more testing.

File: pkg/controller/registry/grpc/source.go (hunk 1 above)
Comment (dinhxuanvu) [reply pkg/controller/registry/grpc/source.go:150]: 👍

**Comment by njhale** [comment]
we've fixed the caching with the first-class github action
